### PR TITLE
[HTML5] Fix target_fps when window loses focus.

### LINKED ISSUE #48061: Vertical synchronization is reactivated when the window regains focus
**Godot version:**
3.3 rc9

**OS/device including version:**
Windows 10, Macbook pro retina 2013, Intel Iris Graphics 6100, Google Chrome
Windows 10, AMD Ryzen 5 2600, GTX 1650, Google Chrome

**Issue description:**
In HTML5 export, i disabled vsync and set FPS target to 30 to make my game more friendly for user with old devices.
Everything work when window's user keep focus.

When you decide to open another tab/window for 10 seconds and you get back on the game tab ; vsync is re-enabled and FPS target is set to 60

**Steps to reproduce:**
1. Make a project with target fps to 0 and vsync disabled, put a button to set FPS target to 30
2. Export to HTML5
3. Run on the browser, check your GPU consumption
4. Open/change tab for 10 seconds
5. Get back to the game tab, vsync is re-enabled, check your GPU consumption

**Minimal reproduction project:**
https://chapatizretro.com/fps/fps.zip

https://chapatizretro.com/fps/fps.html (web version)

### PR DESCRIPTION
**Pull request by Faless**
We don't get updates when the window is unfocused/minimized, so we must detect the situation where the counted ticks start drifting away resulting in more frames drawn than needed.
This commit adds a check to ensure that the target ticks do not drift away more than one second.

Fixes #48061

(the first commit is the usual `RasterizerDummy` fix, the second commit should be cherry-pickable).

### CODE REVIEW

File: platform/javascript/javascript_main.cpp [hunk 1]
```diff
@@ -66,6 +66,11 @@ void main_loop_callback() {
 
 	int target_fps = Engine::get_singleton()->get_target_fps();
 	if (target_fps > 0) {
+		if (current_ticks - target_ticks > 1000000) {
+			// When the window loses focus, we stop getting updates and accumulate delay.
+			// For this reason, if the difference is too big, we reset target tickss to the current ticks.
```
Comment (akien-mga) [inline platform/javascript/javascript_main.cpp:71]:
```suggestion
			// For this reason, if the difference is too big, we reset target ticks to the current ticks.
```

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.

**Comment by akien-mga** [comment]
Cherry-picked for 3.3.1.

**Comment by h0lley** [comment]
I have encountered an issue in my game where I had to change logic after updating Godot:
before, `OS.vsync_enabled` was prioritized over `Engine.target_fps`, that is to say the target FPS was disregarded if vsync was enabled. Now, it is the other way around, so when using vsync, one has to avoid setting target FPS to something other than 0.

Could this be caused by this commit and if so, is this intentional?

**Pull request merged**